config/config.go

```diff
@@ -22,22 +25,34 @@ type SafeConfig struct {
 	C *Config
 }
 
+func init() {
+	configReloadSuccess = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "successful_config_load",
+		Help: "Blackbox exporter config loaded successfully",
+	})
+	prometheus.MustRegister(configReloadSuccess)
+}
+
 func (sc *SafeConfig) ReloadConfig(confFile string) (err error) {
 	var c = &Config{}
 
 	yamlFile, err := ioutil.ReadFile(confFile)
 	if err != nil {
+		configReloadSuccess.Set(0)
```

Comment:
Rather than set at each return, we can use a defer to at the top of the function deal with the error output.

```go
defer func() {
  if err != nil {
    configReloadSuccess.Set(0)
  } else {
    configReloadSuccess.Set(1)
}()
```

Reply:
made this a tad cleaner